services/ymax-planner/src/pending-tx-manager.ts

```diff
@@ -224,12 +226,15 @@ const gmpMonitor: PendingTxMonitor<GmpTx, EvmContext> = {
         fetch: ctx.fetch,
       });
       void liveResultP
-        .then(async found => {
+        .then(async result => {
           log(`${logPrefix} Live mode completed`);
           await resolvePendingTx({
             signingSmartWalletKit: ctx.signingSmartWalletKit,
             txId,
-            status: found ? TxStatus.SUCCESS : TxStatus.FAILED,
+            status: result.found ? TxStatus.SUCCESS : TxStatus.FAILED,
+            ...(result.rejectionReason
+              ? { rejectionReason: result.rejectionReason }
+              : {}),
```

Comment:
minor:
`...({})`  ill result in `rejectReason: undefined` anyway
```suggestion
            rejectReason: result.rejectionReason || undefined
```